charts/generic-service/values.yaml

```diff
@@ -102,3 +102,10 @@ strategy:
   rollingUpdate:
     maxSurge: 100%
     maxUnavailable: 0
+
+env:
```

Comment:
some test data.  By removing each section in turn and then running
```
helm --debug template .
```
I have verified that all 4 scenarios work - both defined, namespace defined, env defined, neither defined.